### DISCUSSION TITLE
RTSP has audio only (no video)

### DISCUSSION THREAD

**Discussion by chall37** [gtxaspec/wz_mini_hacks#155] [Q&A] [answered]
Device: Wyze Cam Pan v2 (FW v4.49.0.97)

Not sure if this is a real issue, or just a configuration problem on my end, but the RTSP stream only appears to contain audio according to VLC.  Stream works as expected in the Wyze app.

![Screen Shot 2022-07-03 at 10 11 58 AM](https://user-images.githubusercontent.com/62623706/177030377-23bc1516-cfc5-41c1-98de-fa42f1f1635e.png)

Relevant Configuration:
![Screen Shot 2022-07-03 at 10 03 51 AM](https://user-images.githubusercontent.com/62623706/177030381-8d4ca18b-d40a-4c68-bf19-045ce3f38d9e.png)

Any help would be appreciated.

**Comment by gtxaspec** [comment]
can you share wz_user.log

**Reply by chall37** [reply]
Here you are: [wz_user.log](https://github.com/gtxaspec/wz_mini_hacks/files/9034434/wz_user.log)

I do see an error about an invalid argument on line 216 (presumably from the rtspserver configuration line above it) as well as some potentially relevant errors on lines 265 & 266.

**Reply by gtxaspec** [reply]
check also wz_post.log
make sure the loopback module was loaded properly

**Reply by chall37** [reply]
I don't see any errors about loading the loopback module, and it appears to be loaded.

[wz_post.log](https://github.com/gtxaspec/wz_mini_hacks/files/9034465/wz_post.log)
[lsmod_and_ps.txt](https://github.com/gtxaspec/wz_mini_hacks/files/9034480/lsmod_and_ps.txt)

**Comment by XBravRevolutionz** [comment]
I found a bug noted online that "Record Sound" has to be disabled under the camera's advanced settings from the app. After disabling that and rebooting, RTSP serves up both audio and video.

**Comment by gtxaspec** [comment, marked as answer]
@cohall-2020 forgive me for not realizing earlier, the firmware version you have on your panv2 may be the problem, its a really really old version.  try 4.49.1.47.

**Reply by chall37** [reply]
That worked, thanks.

**Comment by teredactle** [comment]
@gtxaspec I am having the same issue. Running this on a WCv3 with the RTSP firmware (4.61.0.3). I have turned off in Wyze app the RTSP feature in the firmware, but still I am only getting audio.

>>Is this a known issue **without** a workaround on the RSTP beta firmware from Wyze? I'd prefer to keep this beta FW on the camera as it gives me a quick plan B (and loading the BETA firmware - not available anymore, but obviously I've kept it - is a bit of a PITA).

I think it's worth mentioning in the release notes that _**this hack will NOT work with the WCv3 RTSP beta FW**_ - as many people running this particular FW are looking for a better solution (namely wz_mini_hack) :)

Thank you

edit(s)
-turning "recording audio" to off in the Wyze app did not change anything, still no video on RTSP stream
-when I look at the web interface under diagnostics I see this, not sure what it means (normal/or not):

```
no SD card error
no SD read only error
no file system error
SD card (tf_prepare) error found
Oct 5 10:07:51 syslog: [sdk,0483][sdk_device_mmc_health_test]err: run: tf_prepare failed! exit(3) Oct 5 10:07:51 iCamera: [sdk,0483][sdk_device_mmc_health_test]err: run: tf_prepare failed! exit(3)
SD card health fail error found
```

**Comment by pjuhl2313** [comment]
I'm having an issue with the audio only showing in VLC as well. No video.

**Reply by pjuhl2313** [reply]
Update, the one camera that worked with video and audio RTSP was using firmware 4.36.9.139. The other cams that only played audio with no video had a newer firmware. I reverted those back to 4.36.9.139 and both audio and video work.
